### PR TITLE
Add dns01 challenge post

### PR DESCRIPTION
**Pull request by kosyfrances**

### CODE REVIEW

File: _posts/2020-03-19-letsencrypt-dns01.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+- google cloud
+- ingress
+- letsencrypt
+- cert-manager
+category: blog
+author: kosyanyanwu
+description: Configuring DNS01 Challenge Provider using Google CloudDNS for Let's Encrypt issuer setup with cert-manager
+---
+## Introduction
+[DNS01 challenge](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge) asks you to prove that you control the DNS for your domain name by putting a specific value in a TXT record under that domain name. It is harder to configure than HTTP-01, but can work in scenarios that HTTP-01 can’t. It also allows you to issue wildcard certificates. After Let’s Encrypt gives your ACME client a token, your client will create a TXT record derived from that token and your account key, and put that record at `_acme-challenge.<YOUR_DOMAIN>`. Then Let’s Encrypt will query the DNS system for that record. If it finds a match, you can proceed to issue a certificate!
```
Comment (Kenec) [inline _posts/2020-03-19-letsencrypt-dns01.md]: - As a reader, I may want to know what is hard to configure. A short explanation of the challenge will keep the reader more engaged.

File: _posts/2020-03-19-letsencrypt-dns01.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+This article explains how to set up a ClusterIssuer to use Google CloudDNS to solve [DNS01 ACME challenge](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge). It assumes that your cluster is hosted on Google Cloud Platform (GCP) and that you already have a [domain set up with CloudDNS](https://cloud.google.com/kubernetes-engine/docs/tutorials/configuring-domain-name-static-ip#step_4_configure_your_domain_name_records). It also assumes that you have [cert-manager installed](http://localhost:4000/ingress-gce-letsencrypt/#install-cert-manager) on your cluster.
+
+## Create a Service Account
+Create a [service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts#creating) with `dns.admin` role. This is required for cert-manager to be able to add records to CloudDNS in order to solve the DNS01 challenge.
+
+## Create a Service Account Secret
```
Comment (Kenec) [inline _posts/2020-03-19-letsencrypt-dns01.md]: How about `Create a Secret using the Service Account`?

File: _posts/2020-03-19-letsencrypt-dns01.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    email: you@youremail.com
+    server: https://acme-staging-v02.api.letsencrypt.org/directory
+    privateKeySecretRef:
+      name: letsencrypt-prod-account-key
+    solvers:
+    - dns01:
+        clouddns:
+          # The ID of the GCP project
+          project: $PROJECT_ID
+          # This is the secret used to access the service account
```
Comment (Kenec) [inline _posts/2020-03-19-letsencrypt-dns01.md]: what secret? The comment seems to be irrelevant.

**Review by Kenec — COMMENTED** [review]
Awesome article. I think this will unblock quickly that run into the challenge that you addressed.

File: _posts/2020-03-19-letsencrypt-dns01.md (hunk 1 above)
Comment (kosyfrances) [reply _posts/2020-03-19-letsencrypt-dns01.md]: I think I will remove it instead. I simply got it from the docs directly

File: _posts/2020-03-19-letsencrypt-dns01.md (hunk 2 above)
Comment (kosyfrances) [reply _posts/2020-03-19-letsencrypt-dns01.md]: sounds good

File: _posts/2020-03-19-letsencrypt-dns01.md (hunk 3 above)
Comment (kosyfrances) [reply _posts/2020-03-19-letsencrypt-dns01.md]: the secret you created above .... would be referenced there.

Comment (kosyfrances) [reply _posts/2020-03-19-letsencrypt-dns01.md]: Comment updated
